### DISCUSSION TITLE
I am retrieving data from the database and want to periodically check to assign the user's role. How can I do that?

### DISCUSSION THREAD

**Discussion by Batuhan337tr** [discord-php/DiscordPHP#1129] [Q&A] [answered]
I am using such a code, but since it is in a loop, none of the events are triggered, and the rest of the code is not executing.

Is there a method that would allow me to accomplish this?

![Screenshot_2](https://github.com/discord-php/DiscordPHP/assets/47800213/e9c772ba-da0f-4261-b5ad-85edd24ca30b)

**Comment by key2peace** [comment, marked as answer]
- $allControls can be replaced by $discord-getLoop()
- $sorgu does not contain an array of results, you need to use to use either $sorgu->fetch_array() / $sorgu->fetch_assoc() / $sorgu->fetch_object() in the foreach

**Reply by Batuhan337tr** [reply]
Thanks this is work.
